packages/fast-usdc/src/exos/advancer.js

```diff
@@ -1,19 +1,138 @@
-  assertAllDefined({ feed, statusManager });
-  return zone.exo('Fast USDC Advancer', undefined, {});
+export const prepareAdvancer = (
+  zone,
+  { chainHub, feed, log, statusManager, vowTools: { watch } },
+) => {
+  assertAllDefined({ feed, statusManager, watch });
+
+  const transferHandler = zone.exo(
+    'Fast USDC Advance Transfer Handler',
+    M.interface('TransferHandlerI', {
+      // TODO confirm undefined, and not bigint (sequence)
+      onFulfilled: M.call(M.undefined(), {
+        amount: M.bigint(),
+        destination: ChainAddressShape,
+      }).returns(M.undefined()),
+      onRejected: M.call(M.error(), {
+        amount: M.bigint(),
+        destination: ChainAddressShape,
+      }).returns(M.undefined()),
+    }),
+    {
+      /**
+       * @param {undefined} result TODO confirm this is not a bigint (sequence)
+       * @param {{ destination: ChainAddress; amount: bigint; }} ctx
+       */
+      onFulfilled(result, { destination, amount }) {
+        log(
+          'Advance transfer fulfilled',
+          q({ amount, destination, result }).toString(),
+        );
+      },
+      onRejected(error) {
+        // XXX retry logic?
+        // What do we do if we fail, should we keep a Status?
+        log('Advance transfer rejected', q(error).toString());
+      },
+    },
+  );
+
+  return zone.exoClass(
+    'Fast USDC Advancer',
+    M.interface('AdvancerI', {
+      handleTransactionEvent: M.call(CctpTxEvidenceShape).returns(VowShape),
+    }),
+    /**
+     * @param {{
+     *   localDenom: Denom;
+     *   poolAccount: HostInterface<OrchestrationAccount<{ chainId: 'agoric' }>>;
+     * }} config
+     */
+    config => harden(config),
+    {
+      /**
+       * XXX For now, assume this is invoked when a new entry is
+       * observed via the EventFeed.
+       *
+       * @param {CctpTxEvidence} evidence
+       */
+      handleTransactionEvent(evidence) {
+        // XXX assume EventFeed performs validation checks. Advancer will assert uniqueness.
+
+        const { recipientAddress } = evidence.aux;
+        const { EUD } = addressTools.getQueryParams(recipientAddress).params;
+        if (!EUD) {
+          statusManager.observe(evidence);
+          throw makeError(
+            `recipientAddress does not contain EUD param: ${q(recipientAddress)}`,
+          );
+        }
+
+        // TODO #10391 this can throw, and should make a status update in the catch
+        const { chainId } = chainHub.getChainInfoByAddress(EUD);
+
+        /** @type {DenomAmount} */
+        const requestedAmount = harden({
+          denom: this.state.localDenom,
+          value: BigInt(evidence.tx.amount),
+        });
+
+        // TODO #10391 ensure there's enough funds in poolAccount
+
+        /** @type {ChainAddress} */
+        const destination = harden({
+          chainId,
+          value: EUD,
+          encoding: /** @type {const} */ ('bech32'),
+        });
```

Comment:
probably worth a helper fn

Reply:
Agree, what would we call it? toChainAddress might not capture the `bech32` we're adding. `toBech32ChainAddress` seemed too verbose where it wasn't worth factoring out.

Maybe ChainHub's `getChainInfoByAddress` should do this and instead be `getChainAddress(addr)`?

Reply:
+1 to putting it on `ChainHub` since `chainId` is used here only for this construction.

We shouldn't reuse `getChainInfoByAddress` because this is making a value, not getting one.

consider,
```suggestion
        const destination = chainHub.makeChainAddress(EUD);
```

Reply:
Ok, please see [990a6df](https://github.com/Agoric/agoric-sdk/pull/10406/commits/990a6df71eeb25641ba4c4fb48d6596b4b75ab8e). I decided not to label this a breaking change since `getChainInfoByAddress` has only been around for ~a week, but let me know if I should amend the commit message.